tests/rados/customer_scenarios.py

```diff
@@ -0,0 +1,385 @@
+    mon_db_trim_wait_dur = 1200
+
+    # List to capture the mon db size throughout the duration of the test to check the variations in DB size
+    mon_db_size_list = list()
+    mon_db_size_list.append(get_mondb_size(mon_nodes[0], mon_daemons))
+
+    # Collecting first and last commits to osdmap
+    status = run_ceph_command(node, action="ceph_report")
+    init_commmits = {
+        "osdmap_first_committed": float(status["osdmap_first_committed"]),
+        "osdmap_last_committed": float(status["osdmap_last_committed"]),
+    }
+
+    # creating scenarios where the mon db would be updated with new info
+    if not change_config_for_slow_ops(node=node, action="set", **kwargs):
+        log.error("Failed to setup config changes to generate slow_ops")
+        return False
+
+    # Starting read and write on by creating a test pool .
+    pool_name = "test_pool_ops"
+    if not create_pool(node=node, pool_name=pool_name, crush_rule="stretch_rule"):
+        log.error("failed to create pool to run IO")
+        return False
+
+    mon_db_size_list.append(get_mondb_size(mon_nodes[0], mon_daemons))
+    """rados put and get in a parallel task"""
+    with parallel() as p:
+        p.spawn(do_rados_put, client_node, pool_name, 40)
+        p.spawn(do_rados_get, client_node, pool_name, 40)
+    mon_db_size_list.append(get_mondb_size(mon_nodes[0], mon_daemons))
+
+    # deleting a previously created pool to increase OSD operations and map changes
+    pname = "delete_pool"
+    detete_pool(node=node, pool=pname)
+
+    # Proceeding to reboot 1 OSD from each host to trigger rebalance & Backfill
+    for val in osd_nodes:
+        daemon_info = run_ceph_command(node, action="daemon_info")
+        osd_daemons = [entry for entry in daemon_info if entry["daemon_type"] == "osd"]
+        for osd in osd_daemons:
+            if re.search(osd["hostname"], val.hostname):
+                cmd = f"podman restart {osd['container_id']}"
+                log.info(
+                    f"rebooting osd-{osd['daemon_id']} on host {osd['hostname']}. Command {cmd}"
+                )
+                val.exec_command(sudo=True, cmd=cmd)
+                break
+
+    # Re-weighting the OSd's based on usage to trigger rebalance
+    # todo: Verify re-balancing process on OSD's ( PG movement across cluster)
+    # todo: Add re-balancing based on crush item provided
+    # BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1766702
+    reweight_crush_items(node=node)
+
+    """
+    Waiting for 2 hours for cluster to get to active + clean state.
+    Rationale: during cluster activities like backfill, rebalance, osd map change cause Mon DB to be updated.
+    Hence, we can wait till mon DB updates are completed, after which DB size should be reduced,
+    by trimming the old mappings once the new mappings are added.
+    """
+    end_time = datetime.datetime.now() + datetime.timedelta(seconds=7200)
```

Comment:
No, We cannot speed-up any of the processes. it already makes use of cluster network, and backfilling, recovery threads also would be already increased.

Also, It would not wait for 2 hours if the cluster reaches clean state before that. I feel that 2 hours is less as there is no guarantee that slow_ops would be generated in the given time slot at all. ( Customer bug was explicitly about slow_ops )

But since we are also increasing db size in other methods, we are making those bugs automated. ( need to find better method to definitively generate slow_ops )

But if cluster is still performing operations for long time, we would need at-least some data to make sure DB is not just increasing. ( DB is expected to increase when operations are in progress. Old mappings are removed when operations/ new updates are completed. )